website/src/developing-extensions/custom-deprecations.md

```diff
@@ -0,0 +1,59 @@
+---
+title: Custom deprecations
+---
+
+<div class="text-xs inline-block border border-green-600 text-green-600 bg-green-100 rounded px-1 mb-4">Available in PHPStan 2.1.12</div>
+
+PHPStan allows you adjust behaviour of reflection methods `->isDeprecated()` and `->getDeprecationDescription()` to return custom deprecation information based on e.g. custom PHP attributes. 
```

Comment:
`reflection` should be linked to https://phpstan.org/developing-extensions/reflection